src/gui/DatabaseWidget.cpp

```diff
@@ -755,7 +755,7 @@ void DatabaseWidget::entryActivationSignalReceived(Entry* entry, EntryModel::Mod
 void DatabaseWidget::switchToEntryEdit()
 {
     Entry* entry = m_entryView->currentEntry();
-    Q_ASSERT(entry);
+    
```

Comment:
It's a signal that was directly triggered by QT event. So if you want to catch that before you have to plug in an additional function in between which does nothing else but also returning when there is no valid entry. The code which connects the signal it is this in SearchWidget.cpp:104:

`mx.connect(m_ui->searchEdit, SIGNAL(returnPressed()), SLOT(switchToEntryEdit()));`

Connecting another member function to only catch empty search results seems very superfluous to me when switchToEntryEdit() already handles that case perfectly.